api/src/main/resources/org/openmrs/module/mirebalaisreports/sql/mexicoSuiveSimple.sql

```diff
@@ -0,0 +1,40 @@
+SELECT
+    MAX(cn.name) "Dx",
+    MAX(crt.code) "ICD-10",
+    IF(IFNULL(prev_diag.obs_id, 'null') = 'null', '1', ' ') "1a vez",
+    ROUND(DATEDIFF(e.encounter_datetime, pr.birthdate)/365.25, 1) "Edad",
+    pr.gender "Genero",
+    pa.city_village "Loc"
+FROM patient p
+-- Person
+         INNER JOIN person pr ON p.patient_id = pr.person_id AND pr.voided = 0
+-- Most recent address
+         LEFT OUTER JOIN (SELECT * FROM person_address WHERE voided = 0 ORDER BY date_created DESC) pa
+                         ON p.patient_id = pa.person_id
```

Comment:
Dave suggests

```
LEFT OUTER JOIN person_address a ON a.person_address_id = (select person_address_id from person_address a2 where a2.person_id =  p.person_id and a2.voided = 0
                                                 order by a2.preferred desc, a2.date_created desc limit 1)``
```

Reply:
```
LEFT OUTER JOIN person_address a ON a.person_address_id = (select person_address_id from person_address a2 where a2.person_id =  p.person_id and a2.voided = 0
                                                 order by a2.preferred desc, a2.date_created desc limit 1)
```

is logically equivalent to
```
LEFT OUTER JOIN (SELECT * FROM person_address a WHERE a.voided = 0
                                                ORDER BY a.preferred DESC, a.date_created DESC LIMIT 1)
						ON a.person_id =  p.person_id
```

Which probably is indeed better, since it takes account of "preferred!"

Reply:
NB @dmdesimone

Reply:
Wait, I'm wrong. I'll just use the one you suggested, Dave.